### PR TITLE
Np 1979 update doi request

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: template.yaml [hunk 1]
```diff
@@ -249,22 +241,22 @@ Resources:
...
-                responses: {}
+                responses: { }
                 httpMethod: POST
                 type: AWS_PROXY
               responses:
                 '202':
                   description: Publishing in progress
                   schema:
-                    $ref: '#/definitions/PublishPublicationStatus'
+                    $ref: '#/definitions/PublicationPublishStatus'
```
Comment (axthosarouris) [inline template.yaml:255]: typo

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/DoiRequestService.java [hunk 2]
```diff
@@ -36,15 +45,23 @@
...
 public class DoiRequestService extends ServiceWithTransactions {
 
-    public static final String DOI_REQUEST_NOT_FOUND = "Could not find a Doi Request for Resource: ";
+    public static final String DOI_REQUEST_WAS_NOT_FOUND_ERROR = "DoiRequest was not found";
+    public static final String DOI_REQUEST_NOT_FOUND_FOR_RESOURCE = "Could not find a Doi Request for Resource: ";
+
     private static final Supplier<SortableIdentifier> DEFAULT_IDENTIFIER_PROVIDER = SortableIdentifier::next;
     private static final int DEFAULT_QUERY_RESULT_SIZE = 10_000;
+    public static final String UPDATE_DOI_REQUEST_STATUS_CONDITION_FAILURE_MESSAGE =
+        "Could not update doi request status. Updating Doi request status for "
```
Comment (brinxmat) [inline publication-commons/src/main/java/no/unit/nva/publication/service/impl/DoiRequestService.java:62]: Capitalise DOI in all cases

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/DoiRequestService.java [hunk 3]
```diff
@@ -36,15 +45,23 @@
...
+        + "not published resources is not possible";
```
Comment (brinxmat) [inline publication-commons/src/main/java/no/unit/nva/publication/service/impl/DoiRequestService.java:63]: unpublished

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/DoiRequestService.java [hunk 4]
```diff
@@ -127,11 +143,125 @@ public DoiRequest getDoiRequest(UserInstance userInstance, SortableIdentifier id
...
+                                  + "#data.#status = :status, "
+                                  + "#data.#modifiedDate = :modifiedDate,"
+                                  + "#PK1 = :PK1 ,"
+                                  + "#SK1 = :SK1 ,"
+                                  + "#PK2 = :PK2 ,"
+                                  + "#SK2 = :SK2 ";
+
+        String conditionExpression = "#data.#resourceStatus = :publishedStatus";
+
+        Map<String, String> expressionAttributeNames = Map.of(
```
Comment (brinxmat) [inline publication-commons/src/main/java/no/unit/nva/publication/service/impl/DoiRequestService.java:216]: Can we extract these to methods?

**Review by brinxmat — APPROVED** [review]

**Pull request merged**